dali/aux/optical_flow/turing_of/optical_flow_turing_test.cc

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <dali/aux/optical_flow/turing_of/optical_flow_turing.h>
+#include <dali/test/dali_test_config.h>
+#include <dali/util/cuda_utils.h>
+#include <cuda_runtime.h>
+#include <gtest/gtest.h>
+#include <fstream>
+#include <vector>
+#include <string>
+
+namespace dali {
+namespace optical_flow {
+namespace testing {
+
+namespace {
+constexpr float kFlowVectorEpsilon = 1.f / 32;
+}
+
+TEST(OpticalFlowTuringTest, DecodeFlowVectorTest) {
+  using std::vector;
+  vector<int16_t> test_data = {101, -32376, 676, 3453, -23188};
+  vector<float> ref_data = {3.15625f, -12.25f, 21.125f, 107.90625f, -299.375f};
+  for (size_t i = 0; i < test_data.size(); i++) {
+    EXPECT_NEAR(ref_data[i], kernel::decode_flow_component(test_data[i]), kFlowVectorEpsilon);
+  }
+}
+
+
+TEST(OpticalFlowTuringTest, CudaDecodeFlowVectorTest) {
+  using std::ifstream;
+  using std::string;
+  using std::vector;
+  auto test_data_path = dali::testing::dali_extra_path() + "/db/optical_flow/flow_vector_decode/";
+  ifstream infile(test_data_path + "flow_vector_test_data.txt"),
```

Comment:
Is the data really so large to mandate the use of a data file instead of embedding the data in the test file?

Reply:
10000 samples, keeping it if file seems more convenient to me